crates/core/src/operations/drop_column.rs

```diff
@@ -0,0 +1,213 @@
+//! Drop a column from the table
+
+use delta_kernel::schema::DataType;
+use delta_kernel::schema::StructType;
+use futures::future::BoxFuture;
+use std::collections::HashMap;
+
+use itertools::Itertools;
+use sqlparser::dialect::GenericDialect;
+use sqlparser::parser::Parser;
+
+use super::transaction::{CommitBuilder, CommitProperties};
+
+use crate::kernel::StructField;
+use crate::logstore::LogStoreRef;
+use crate::protocol::DeltaOperation;
+use crate::table::state::DeltaTableState;
+use crate::{DeltaResult, DeltaTable, DeltaTableError};
+
+/// Add new columns and/or nested fields to a table
+pub struct DropColumnBuilder {
+    /// A snapshot of the table's state
+    snapshot: DeltaTableState,
+    /// Fields to drop from the schema
+    fields: Option<Vec<String>>,
+    /// Raise if constraint doesn't exist
+    raise_if_not_exists: bool,
+    /// Delta object store for handling data files
+    log_store: LogStoreRef,
+    /// Additional information to add to the commit
+    commit_properties: CommitProperties,
+}
+
+impl super::Operation<()> for DropColumnBuilder {}
+
+impl DropColumnBuilder {
+    /// Create a new builder
+    pub fn new(log_store: LogStoreRef, snapshot: DeltaTableState) -> Self {
+        Self {
+            snapshot,
+            log_store,
+            raise_if_not_exists: true,
+            fields: None,
+            commit_properties: CommitProperties::default(),
+        }
+    }
+
+    /// Specify the fields to be added
+    pub fn with_fields(mut self, fields: impl IntoIterator<Item = String> + Clone) -> Self {
+        self.fields = Some(fields.into_iter().collect());
+        self
+    }
+    /// Additional metadata to be added to commit info
+    pub fn with_commit_properties(mut self, commit_properties: CommitProperties) -> Self {
+        self.commit_properties = commit_properties;
+        self
+    }
+
+    /// Specify if you want to raise if the specified column does not exist
+    pub fn with_raise_if_not_exists(mut self, raise: bool) -> Self {
+        self.raise_if_not_exists = raise;
+        self
+    }
```

Comment:
Question out of curiosity: What would be the use case or reason to allow drop column commits for columns that are missing from the schema?

Reply:
Let's say I drop 10 columns, and one is a typo, it could still be useful for the commit to pass.

I probably need to add another check though to check if the schema actually changed, otherwise the commits are pointless